src/UI/CanvasController.java

```diff
@@ -0,0 +1,1008 @@
+					+ "red: " + intensity.r
+					+ "green: " + intensity.g
+					+ "blue: " + intensity.b
+				);
+
+	    		logger.log(
+					ModuleID.UI,
+					LogLevel.INFO,
+					"Performing Color Change"
+				);
+
+	    		// Perform Color Change
+	    		ProcessingFactory
+	    			.getProcessor()
+	    			.colorChange(intensity);
+
+	    		logger.log(
+					ModuleID.UI,
+					LogLevel.INFO,
+					"Color Change Performed"
+				);
+
+	    		// Unselect currently selected object
+	        	updateSelectedPixels(null);
+	    	}
+	    	else // Not in cursor Mode or object not selected
+	    		logger.log(
+					ModuleID.UI,
+					LogLevel.INFO,
+					"Not in Cursor Mode or Object not selected"
+				);
+    	}
+    }
+
+    /**
+     * This is used for constructing a small square around the selected point
+     * since it is possible that the user may make slight errors in choosing
+     * a point. We assume that the user selected the entire small square, not
+     * just a single point
+     */
+    private final int[] selRange = new int[] {-1, 0, 1};
+
+    /**
+     * When a point in the canvas in clicked, then this method is called.
+     * If the UI is in Cursor Mode, we pass a small square around the
+     * selected point to the processor so that we may get the object
+     * corresponding to the selected pixels. We highlight the pixels of the
+     * object by representing them with a specific color
+     *
+     * @param canvasMouseClickEvent Mouse Click Event which lead to this
+     * function being called
+     */
+    @FXML
+    void clickCanvas(MouseEvent canvasMouseClickEvent) {
+
+    	synchronized(this) {
+
+	    	// Get the logger
+	    	ILogger logger = LoggerFactory.getLoggerInstance();
+
+	    	// If current mode is not the Cursor Mode, return
+	    	if(currentMode != CurrentMode.CURSOR_MODE) {
+	    		logger.log(
+					ModuleID.UI,
+					LogLevel.INFO,
+					"Unable to Select Object: Not in Cursor Mode"
+				);
+		    	return;
+	    	}
+
+	    	// Get approximate selected row and column coordinate
+	    	int rowCoord = (int) Math.round(canvasMouseClickEvent.getY());
+	    	int colCoord = (int) Math.round(canvasMouseClickEvent.getX());
+
+	    	// selPosition would represent a small region around the
+	    	// approximate selected position
+	    	ArrayList<Position> selPosition = new ArrayList<Position>();
+
+	    	// Construct selPosition by building a small square as the
+	    	// selected region
+	    	for(int r : selRange) {
+	    		for(int c : selRange) {
+	    			int row = rowCoord + r;
+	    			int col = colCoord + c;
+
+	    			if(
+						0 <= row && row < dimension.numRows
+						&&
+						0 <= col && col < dimension.numCols
+	    			)
+	    				selPosition.add(new Position(row, col));
+	    		}
+	    	}
+
+	    	logger.log(
+				ModuleID.UI,
+				LogLevel.INFO,
+				"Getting Selected Object's Pixels from Processing Module"
+			);
+
+	    	// Get Selected Object Pixels from the Processor
+	    	ArrayList<Pixel> objectPixels =
```

Comment:
Currently we are returning ArrayList<Position> instead of ArrayList<Pixel>, if you want the later, we need to change it first.

Reply:
I discussed regarding this with Sakshi and she agreed and accepted to update in the processing module. So we proceeded with this design.